File: MASTER DSA/TODO/LONG PRESSED NAME.java
class Solution {
    public boolean isLongPressedName(String name, String typed) {

        int i = 0;
        int j = 0;

        while (i < name.length() && j < typed.length()) {
            if (name.charAt(i) == typed.charAt(j)) {
                i++;
                j++;
            } else if (i >= 1 && name.charAt(i - 1) == typed.charAt(j)) {
                j++;
            } else {
                return false;
            }
        }

        while (j < typed.length()) {
            if (name.charAt(i - 1) == typed.charAt(j)) {
                j++;
            } else {
                return false;
            }
        }

        return i == name.length() && j == typed.length();
    }
}

File: MASTER_DSA/961-long-pressed-name/long-pressed-name.py
class Solution:
    def isLongPressedName(self, name: str, typed: str) -> bool:
        i = 0  # to iterated name 
        j = 0  # to iterate typed

        while i < len(name) and j < len(typed) :
            if name[i] == typed[j]:
                i+=1
                j+=1
            elif name[i] != typed[j] and j>0 and typed[j-1] == typed[j]:
                j+=1
            else :
                return False

        # if last character repeated 
        while j < len(typed) and typed[j-1] == typed[j]:
            j+=1
        
        if i != len(name) or j!= len(typed):
            return False

        return True
        